Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -672,8 +696,78 @@ - (NSInteger)readIntegerFromDefaults:(NSString *)key {
     return NSNotFound;
 }
 
-- (NSString *)prefsFile {
-    return [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] stringByAppendingPathComponent:BRANCH_PREFS_FILE];
-}
++ (NSString *)prefsFile_deprecated
+    {
+    NSString * path =
+        [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES)
+            firstObject]
+                stringByAppendingPathComponent:BRANCH_PREFS_FILE];
+    return path;
+    }
+
++ (NSURL*) URLForBranchDirectory
+    {
+    NSError *error = nil;
+    NSURL *URL =
+        [[NSFileManager defaultManager]
+            URLForDirectory:NSApplicationSupportDirectory
+            inDomain:NSUserDomainMask
+            appropriateForURL:nil
+            create:YES
+            error:&error];
+    if (error)
+        {
+        NSLog(@"Error creating URLForPrefsDirectory: %@.", error);
+        return nil;
+        }
+    URL = [URL URLByAppendingPathComponent:@"io.branch"];
+    [[NSFileManager defaultManager]
+        createDirectoryAtURL:URL
+        withIntermediateDirectories:YES
+        attributes:nil
+        error:&error];
+    if (error)
+        {
+        NSLog(@"Error creating URLForPrefsDirectory: %@.", error);
+        return nil;
+        }
+    return URL;
+    }
+
++ (NSURL*) URLForPrefsFile
+    {
+    NSURL *URL = [self URLForBranchDirectory];
+    URL = [URL URLByAppendingPathComponent:BRANCH_PREFS_FILE];
+    return URL;
+    }
+
++ (void) moveOldPrefsFile
+    {
+    NSURL *oldURL = [NSURL fileURLWithPath:self.prefsFile_deprecated];
+    NSURL *newURL = [self URLForBranchDirectory];
+
+    NSError *error = nil;
+    [[NSFileManager defaultManager]
+        moveItemAtURL:oldURL
+        toURL:newURL
+        error:&error];
+
+    if (error && error.code != NSFileNoSuchFileError)
+        {
+        if (error.code == NSFileWriteFileExistsError)
+            {
+            [[NSFileManager defaultManager]
+                removeItemAtURL:oldURL
+                error:&error];
+            }
+        else
+            NSLog(@"Error moving prefs file: %@.", error);
+        }
+    }
+
++ (void) initialize
+    {
+    [self moveOldPrefsFile];
```

Comment:
The ```+initialize``` method is called once after the class is loaded and before the first instance method of the class is called.  So there is little overhead.